src/contracts/yas_pool.cairo

```diff
@@ -0,0 +1,49 @@
+#[starknet::interface]
+trait IYASPool<TContractState> {
+    fn initialize(ref self: TContractState);
+}
+
+#[starknet::contract]
+mod YASPool {
+    use super::IYASPool;
+    use starknet::ContractAddress;
+
+    use yas::libraries::tick::Tick;
+    use yas::numbers::signed_integer::{i32::i32, integer_trait::IntegerTrait};
+
+    #[storage]
+    struct Storage {
+        factory: ContractAddress,
+        token_0: ContractAddress,
+        token_1: ContractAddress,
+        fee: u32,
+        liquidity_per_tick: u128,
+    }
+
+    #[constructor]
+    fn constructor(
+        ref self: ContractState,
+        factory: ContractAddress,
+        token_0: ContractAddress,
+        token_1: ContractAddress,
+        fee: u32,
+        tick_spacing: i32,
+    ) {
+        self.factory.write(factory);
+        self.token_0.write(token_0);
+        self.token_1.write(token_1);
+        self.fee.write(fee);
+
+        //TODO: temporary component syntax
+        let state = Tick::unsafe_new_contract_state();
+        self
+            .liquidity_per_tick
+            .write(Tick::TickImpl::tick_spacing_to_max_liquidity_per_tick(@state, tick_spacing));
+    }
+
+    #[external(v0)]
+    impl YASPoolImpl of IYASPool<ContractState> {
+        fn initialize(ref self: ContractState) { // TODO: implement
```

Comment:
this method is from `IUniswapV3PoolActions`?

Reply:
This method belongs to [UniswapV3Pool](https://github.com/Uniswap/v3-core/blob/d8b1c635c275d2a9450bd6a78f3fa2484fef73eb/contracts/UniswapV3Pool.sol#L271). I created the interface with this method to generate the dispatcher. We need the latter to conduct tests

Reply:
The implementation of the `initialize` function will be resolved in another PR

Reply:
nice, I was asking because there wasn't even the method's signature, but it was to understand where it was coming from